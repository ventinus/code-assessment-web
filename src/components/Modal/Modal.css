.modal {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-color: rgba(0, 0, 0, 0.3);
  transform: translateY(-100%);
  transition: transform 0.3s ease-in-out 0s; }
  .modal.is-open {
    transform: translateY(0); }

.modal__inner {
  background-color: white;
  height: 100%; }
  @media (min-width: 768px) {
    .modal__inner {
      position: absolute;
      top: 50%;
      left: 50%;
      transform: translate(-50%, -50%);
      height: auto;
      border-radius: 10px;
      overflow: hidden; } }

.modal__inner__close {
  position: absolute;
  top: 20px;
  right: 20px;
  width: 20px;
  height: 20px; }
  .modal__inner__close:before, .modal__inner__close:after {
    content: '';
    position: absolute;
    background-color: #d8d8d8;
    height: 2px;
    width: 25px;
    top: 50%;
    left: 50%; }
  .modal__inner__close:before {
    transform: translate(-50%, -50%) rotate(-45deg); }
  .modal__inner__close:after {
    transform: translate(-50%, -50%) rotate(45deg); }
  @media (min-width: 768px) {
    .modal__inner__close {
      width: 30px;
      height: 30px;
      top: 30px;
      right: 30px; }
      .modal__inner__close:before, .modal__inner__close:after {
        width: 40px; } }
