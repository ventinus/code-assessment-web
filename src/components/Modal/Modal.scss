@import '../../scss/variables/mixins.scss';
@import '../../scss/variables/variables.scss';

.modal {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-color: rgba($black, 0.3);
  transform: translateY(-100%);
  transition: default-transition(transform);

  &.is-open {
    transform: translateY(0);
  }
}

.modal__inner {
  background-color: $white;
  height: 100%;

  @include breakpoint(md) {
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    height: auto;
    border-radius: 10px;
    overflow: hidden;
  }
}

.modal__inner__close {
  position: absolute;
  top: 15px;
  right: 15px;
  width: 30px;
  height: 30px;
  z-index: 10;

  &:before,
  &:after {
    content: '';
    position: absolute;
    background-color: $grey-4;
    height: 2px;
    width: 25px;
    top: 50%;
    left: 50%;
  }

  &:before {
    transform: translate(-50%, -50%) rotate(-45deg);
  }
  &:after {
    transform: translate(-50%, -50%) rotate(45deg);
  }

  @include breakpoint(md) {
    width: 40px;
    height: 40px;

    &:before,
    &:after { width: 40px; }
  }
}
