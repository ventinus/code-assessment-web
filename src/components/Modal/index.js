import React from 'react'
import PropTypes from 'prop-types'

import './Modal.css'

const Modal = ({ isOpen = false, toggleModal, children }) => (
  <div className={`modal ${isOpen ? 'is-open' : ''}`}>
    <div className="modal__inner">
      <button className="modal__inner__close" onClick={toggleModal}>
        <span className="is-visually-hidden">Close Modal</span>
      </button>
      {children}
    </div>
  </div>
)

Modal.propTypes = {
  isOpen: PropTypes.bool,
  toggleModal: PropTypes.func.isRequired,
  children: PropTypes.any,
}

export default Modal
