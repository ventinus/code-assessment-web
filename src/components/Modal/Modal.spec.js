import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import Modal from '.'

const setup = ( props = {} ) => {
  const toggleModal = jest.fn()

  const component = shallow(
    <Modal {...props} toggleModal={toggleModal}>Modal inside</Modal>
  )

  return {
    component,
    toggleModal,
    button: component.find('button'),
  }
}

describe('Modal component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Modal toggleModal={jest.fn()}>children</Modal>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should handle toggleModal', () => {
    const { button, toggleModal } = setup()
    button.simulate('click')
    expect(toggleModal).toHaveBeenCalled()
  })

  it('should not include is-open class', () => {
    const { component } = setup()
    expect(component.props().className).toBe('modal ')
  })

  describe('when open', () => {
    it('should include is-open class', () => {
      const { component } = setup({isOpen: true})
      expect(component.props().className).toBe('modal is-open')
    })
  })
})
