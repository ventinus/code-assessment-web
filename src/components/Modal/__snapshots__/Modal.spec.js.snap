// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component renders correctly 1`] = `
<div
  className="modal "
>
  <div
    className="modal__inner"
  >
    <button
      className="modal__inner__close"
      onClick={[MockFunction]}
    >
      <span
        className="is-visually-hidden"
      >
        Close Modal
      </span>
    </button>
    children
  </div>
</div>
`;
